File: sdks/sdkwork-craw-chat-sdk/sdkwork-craw-chat-sdk-flutter/composed/lib/src/rtc_module.dart
import 'package:backend_sdk/backend_sdk.dart';

import 'builders.dart';
import 'context.dart';
import 'types.dart';

class CrawChatRtcModule {
  final CrawChatSdkContext context;

  CrawChatRtcModule(this.context);

  Future<RtcSession?> create(CreateRtcSessionRequest body) {
    return context.backendClient.rtc.createRtcSession(body);
  }

  Future<RtcSession?> invite(
    String rtcSessionId,
    InviteRtcSessionRequest body,
  ) {
    return context.backendClient.rtc.inviteRtcSession(rtcSessionId, body);
  }

  Future<RtcSession?> accept(
    String rtcSessionId,
    UpdateRtcSessionRequest body,
  ) {
    return context.backendClient.rtc.acceptRtcSession(rtcSessionId, body);
  }

  Future<RtcSession?> reject(
    String rtcSessionId,
    UpdateRtcSessionRequest body,
  ) {
    return context.backendClient.rtc.rejectRtcSession(rtcSessionId, body);
  }

  Future<RtcSession?> end(
    String rtcSessionId,
    UpdateRtcSessionRequest body,
  ) {
    return context.backendClient.rtc.endRtcSession(rtcSessionId, body);
  }

  Future<RtcSignalEvent?> postSignal(
    String rtcSessionId,
    PostRtcSignalRequest body,
  ) {
    return context.backendClient.rtc.postRtcSignal(rtcSessionId, body);
  }

  Future<RtcSignalEvent?> postJsonSignal(
    String rtcSessionId, {
    required String signalType,
    required CrawChatPostJsonSignalOptions options,
  }) {
    return postSignal(
      rtcSessionId,
      CrawChatBuilders.jsonRtcSignal(
        signalType: signalType,
        options: options,
      ),
    );
  }

  Future<RtcParticipantCredential?> issueParticipantCredential(
    String rtcSessionId,
    IssueRtcParticipantCredentialRequest body,
  ) {
    return context.backendClient.rtc.issueRtcParticipantCredential(
      rtcSessionId,
      body,
    );
  }

  Future<RtcRecordingArtifact?> getRecordingArtifact(String rtcSessionId) {
    return context.backendClient.rtc.getRtcRecordingArtifact(rtcSessionId);
  }
}

File: sdks/sdkwork-craw-chat-sdk/sdkwork-craw-chat-sdk-typescript/composed/src/rtc-module.ts
import { buildJsonRtcSignalRequest } from './builders.js';
import type {
  CreateRtcSessionRequest,
  InviteRtcSessionRequest,
  IssueRtcParticipantCredentialRequest,
  PostJsonRtcSignalOptions,
  PostRtcSignalRequest,
  RtcParticipantCredential,
  RtcRecordingArtifact,
  RtcSession,
  RtcSignalEvent,
  UpdateRtcSessionRequest,
} from './types.js';
import type { CrawChatSdkContext } from './sdk-context.js';

export class CrawChatRtcModule {
  constructor(private readonly context: CrawChatSdkContext) {}

  create(body: CreateRtcSessionRequest): Promise<RtcSession> {
    return this.context.backendClient.rtc.createRtcSession(body);
  }

  invite(
    rtcSessionId: string | number,
    body: InviteRtcSessionRequest,
  ): Promise<RtcSession> {
    return this.context.backendClient.rtc.inviteRtcSession(rtcSessionId, body);
  }

  accept(
    rtcSessionId: string | number,
    body: UpdateRtcSessionRequest,
  ): Promise<RtcSession> {
    return this.context.backendClient.rtc.acceptRtcSession(rtcSessionId, body);
  }

  reject(
    rtcSessionId: string | number,
    body: UpdateRtcSessionRequest,
  ): Promise<RtcSession> {
    return this.context.backendClient.rtc.rejectRtcSession(rtcSessionId, body);
  }

  end(
    rtcSessionId: string | number,
    body: UpdateRtcSessionRequest,
  ): Promise<RtcSession> {
    return this.context.backendClient.rtc.endRtcSession(rtcSessionId, body);
  }

  postSignal(
    rtcSessionId: string | number,
    body: PostRtcSignalRequest,
  ): Promise<RtcSignalEvent> {
    return this.context.backendClient.rtc.postRtcSignal(rtcSessionId, body);
  }

  postJsonSignal(
    rtcSessionId: string | number,
    signalType: string,
    options: PostJsonRtcSignalOptions,
  ): Promise<RtcSignalEvent> {
    return this.postSignal(
      rtcSessionId,
      buildJsonRtcSignalRequest(signalType, options),
    );
  }

  issueParticipantCredential(
    rtcSessionId: string | number,
    body: IssueRtcParticipantCredentialRequest,
  ): Promise<RtcParticipantCredential> {
    return this.context.backendClient.rtc.issueRtcParticipantCredential(
      rtcSessionId,
      body,
    );
  }

  getRecordingArtifact(
    rtcSessionId: string | number,
  ): Promise<RtcRecordingArtifact> {
    return this.context.backendClient.rtc.getRtcRecordingArtifact(rtcSessionId);
  }
}
